File: GFG/Mathematics/check_prime.cpp
#include<iostream>
using namespace std;

bool isPrime(int q){
    if(q%2==0 or q%3==0){
        return false;
    }
    for(int i=5;i*i<=q;i++){
        if (q%i==0){
            return false;
        }
        i++;
    }
    return true;
}

int main (){
    cout<<"Enter number to check: "<<endl;
    int x;
    cin>>x;
    cout<<isPrime(x);
    return 0;
}

File: GFG/Mathematics/check_prime.py
def isPrime(z):
    if (z%2==0):
        return False
    elif (z%3==0):
        return False
    i=5
    while (i*i<=z):
        if z%i==0:
            return False
        i+=1
    return True

q=int(input("Enter number to check for prime: "))
print(isPrime(q))